pkg/jobs/canary/canary_jobs.go

```diff
@@ -333,52 +334,25 @@ func SyncCanaryJob(canary v1.Canary) error {
 func SyncCanaryJobs() {
 	logger.Debugf("Syncing canary jobs")
 
-	jobHistory := models.NewJobHistory("CanarySync", "canary", "").Start()
-	_ = db.PersistJobHistory(jobHistory)
-	defer func() { _ = db.PersistJobHistory(jobHistory.End()) }()
-
 	canaries, err := db.GetAllCanariesForSync()
 	if err != nil {
 		logger.Errorf("Failed to get canaries: %v", err)
-		jobHistory.AddError(err.Error())
 		return
 	}
 
 	for _, c := range canaries {
-		canary, err := c.ToV1()
-		if err != nil {
-			logger.Errorf("Error parsing canary[%s]: %v", c.ID, err)
-			jobHistory.AddError(err.Error())
-			continue
-		}
-
-		if len(canary.Status.Checks) == 0 && len(canary.Spec.GetAllChecks()) > 0 {
-			logger.Infof("Persisting %s as it has no checks", canary.Name)
-			pkgCanary, _, _, err := db.PersistCanary(*canary, canary.Annotations["source"])
-			if err != nil {
-				logger.Errorf("Failed to persist canary %s: %v", canary.Name, err)
-				jobHistory.AddError(err.Error())
-				continue
-			}
```

Comment:
Since these canaries are always fetched from database, no persistence logic is needed

Reply:
@yashmehrotra we were also saving checks from the canary spec in here. Now, checks aren't being saved.

Reply:
@yashmehrotra I'm not talking about component checks. If you create a canary from the UI then the checks are never saved.